e2e-tests/pipeline_validation.sh

```diff
@@ -135,21 +151,37 @@ function fhir_source_query() {
       enc_obs_query_param="?subject.given=Alberta625"
   fi
   
-  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
-    "${FHIR_SERVER_URL}/Patient${patient_query_param}" 2>/dev/null >>"${HOME_PATH}/${PARQUET_SUBDIR}/patients.json"
+  if [[ -n ${JDBC_MODE_HAPI} ]]; then
+      patient_query_param="?_summary=count"
+      enc_obs_query_param="?_summary=count"
+
+      curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
```

Comment:
Why do we need to separate these `curl` commands depending on whether we are querying HAPI or not? I mean, there seems a lot of commonality, can't we refactor these instead of repeating these `curl` commands in both if/else blocks?